### ISSUE TITLE
Unnecessary direct unwraps on Option and Result are not warned/fixed

### ISSUE THREAD

**Issue by pksunkara** [rust-lang/rust-clippy#10352]
### Summary

When we have direct unwraps on an `Option` or `Result` like the following:

```rust
let val = Some(1).unwrap();
let val = Ok(1).unwrap();
let val = Err(1).unwrap_err();
let val = Ok(1).expect(...);
```

The lint remove the unnecessary unwrap/expect and unnecessary wraps from the above lines.

```rust
let val = 1;
let val = 1;
let val = 1;
let val = 1;
```

### Lint Name

unnecessary_unwrap

### Reproducer

I tried this code:

```rust
let val = Some(1).unwrap();
```

I expected to see this happen:

```rust
let val = 1;
```

Instead, this happened:

Nothing changed.

**Comment by pksunkara** [comment]
```rust
fn main() {
    let val = Ok(1).unwrap();
    let val = Err(1).unwrap_err();
    let val = Ok(1).expect(...);
}
```

Doesn't seem to compile.

```
Compiling playground v0.0.1 (/playground)
error: unexpected token: `...`
 --> src/main.rs:4:28
  |
4 |     let val = Ok(1).expect(...);
  |                            ^^^
  |
help: use `..` for an exclusive range
  |
4 |     let val = Ok(1).expect(..);
  |                            ~~
help: or `..=` for an inclusive range
  |
4 |     let val = Ok(1).expect(..=);
  |                            ~~~

error[[E0586]](https://doc.rust-lang.org/stable/error_codes/E0586.html): inclusive range with no end
 --> src/main.rs:4:28
  |
4 |     let val = Ok(1).expect(...);
  |                            ^^^ help: use `..` instead
  |
  = note: inclusive ranges must be bounded at the end (`..=b` or `a..=b`)

For more information about this error, try `rustc --explain E0586`.
error: could not compile `playground` due to 2 previous errors
```
